tap_iterable/sync.py

```diff
@@ -7,8 +7,43 @@
 import singer.metrics as metrics
 from singer import metadata
 from singer import Transformer
+from tap_iterable.streams import STREAMS
 
-logger = singer.get_logger()
+LOGGER = singer.get_logger()
+
+def stream_is_selected(mdata):
+    return mdata.get((), {}).get('selected', False)
+
+def get_selected_streams(catalog):
+    selected_stream_names = []
+    for stream in catalog.streams:
+        mdata = metadata.to_map(stream.metadata)
+        if stream_is_selected(mdata):
+            selected_stream_names.append(stream.tap_stream_id)
+    return selected_stream_names
+
+def sync(client, catalog, state):
+    selected_stream_names = get_selected_streams(catalog)
+
```

Comment:
can we change this logic to use the inbuilt get_selected_streams of the catalog object?